### PR TITLE
Updated dependencies and files with Flutter 2.10

### LINKED ISSUE #149: Fix and enable linter rules
**Is your feature request related to a problem? Please describe.**

Yes, In flutter-lints dependency there are so many linter rules that I disabled in PR #148 

Now here are the checklists of the lint rules that are disabled (enabled rules are marked as checked) :

- [x] always_declare_return_types
- [x] avoid_print
- [x] avoid_unnecessary_containers
- [x] prefer_if_elements_to_conditional_expressions
- [x] prefer_const_constructors
- [x] prefer_const_declarations
- [x] unnecessary_string_interpolations
- [x] sized_box_for_whitespace
- [x] prefer_single_quotes
- [x] prefer_const_constructors_in_immutables
- [x] use_key_in_widget_constructors
- [x] prefer_function_declarations_over_variables
- [x] unnecessary_string_escapes
- [x] avoid_function_literals_in_foreach_calls
- [x] prefer_interpolation_to_compose_strings
- [ ] unnecessary_parenthesis
- [x] sort_constructors_first
- [x] always_put_required_named_parameters_first
- [ ] avoid_classes_with_only_static_members
- [ ] avoid_redundant_argument_values
- [x] join_return_with_assignment
- [ ] non_constant_identifier_names
- [ ] constant_identifier_names

### PR DESCRIPTION
**Pull request by ItsAdityaKSingh**
Describe the changes you have made in this PR - Updated dependencies, sorted them, and added a updates for successful runs in Flutter version 2.10+

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -3,6 +3,10 @@ include: package:flutter_lints/flutter.yaml
 analyzer:
   exclude: [lib/generated_plugin_registrant.dart]
 
+language:
```
Comment (manjotsidhu) [inline analysis_options.yaml:6]: Please explain these changes.

Comment (ItsAdityaKSingh) [reply analysis_options.yaml:6]: These ensure that code specifies generic type arguments and make sure that code is maintainable with improved performance. They were by default mentioned in many analysis files, so I thought of using them here too. You may want to read [this](https://github.com/dart-lang/language/blob/master/resources/type-system/strict-raw-types.md) to learn more.

Comment (manjotsidhu) [reply analysis_options.yaml:6]: How it fixes the issue #149 ? Your aim for this PR is to fix #149 or to do something irrelevant ?

Comment (ItsAdityaKSingh) [reply analysis_options.yaml:6]: It tries to add more customary linting checks and analysis options for a better analysis, that's why I mentioned partially fixes 😅

**Comment by ItsAdityaKSingh** [comment]
@manjotsidhu The tests that were passing earlier are failing now with exceptions. Any idea why?

**Comment by ItsAdityaKSingh** [comment]
@manjotsidhu, please review these changes.

**Comment by manjotsidhu** [comment]
@aman-singh7 Can you review these changes, I am busy

**Comment by aman-singh7** [comment]
@ItsAdityaKSingh Please rebase your branch changes with the master.

**Comment by aman-singh7** [comment]
@ItsAdityaKSingh if you are using Mac. Can you also update the Podfile?

**Comment by ItsAdityaKSingh** [comment]
@aman-singh7 There are no such updates for the Podfile while running the app with latest changes.

**Comment by aman-singh7** [comment]
@ItsAdityaKSingh Have you tried to build it for ios?

**Comment by aman-singh7** [comment]
@manjotsidhu I have tested the app and there are no such issues. I think we can go with it.

**Review by manjotsidhu — APPROVED** [review]

**Pull request merged**